docs/guides/es2k/deploying-p4-programs.md

```diff
@@ -0,0 +1,139 @@
+
+```bash
+root@mev-imc:/etc/dpcp/package# ls -lrt /etc/dpcp/package/
+total 2364
+-rw-r--r-- 1 root root  963032 Jan  1 04:56 simple_l2_demo.pkg
+-rw-r--r-- 1 root root 1450456 Jun  8  2023 e2100-default-1.0.3.0.pkg
+drwxr-xr-x 2 root root       0 Jun  8  2023 runtime_files
+drwxr-xr-x 3 root root       0 Jun  8  2023 ref_pkg
+lrwxrwxrwx 1 root root      25 Jun  8  2023 default_pkg.pkg -> e2100-default-1.0.3.0.pkg
+root@mev-imc:/etc/dpcp/package# cp simple_l2_demo.pkg default_pkg.pkg
+```
+
+### 2.3 Start the IMC
+
+Run the IMC start-up script.
+
+```bash
+root@mev-imc:~# /etc/init.d/run_default_init_app
+```
+
+## 3. Deploying on Software Control Plane (infrap4d)
+
+On ACC, use `tdi_pipeline_builder` available under `/opt/p4/p4-cp-nws/bin` to
+generate forwarding pipeline binary.
+
+Note: We will be using artifacts generated by following
+[Compilling P4 programs guide](https://github.com/nupuruttarwar/networking-recipe/blob/main/docs/guides/es2k/compiling-p4-programs.md).
+
+### 3.1 Prepare configuration files
+
+Create tofino.bin
+
+```bash
+export OUTPUT_DIR=/usr/share/mev_reference_p4_files/simple_l3_l4_pna
+cd $OUTPUT_DIR
+touch tofino.bin
+```
+
+Handcraft the configuration file `/usr/share/stratum/es2k_skip_p4.conf` with
+following parameters:
+
+- `pcie_bdf`
+
+   Get PCI BDF of LAN Control Physical Function (CPF) device with device
+   ID 1453 on ACC
+
+   ```bash
+   lspci | grep 1453
+   00:01.6 Ethernet controller: Intel Corporation Device 1453 (rev 11)
+   ```
+
+   Value of `pcie_bdf` should be `00:01.6`
+
+- `iommu_grp_num`
+
+   Get the iommu group number
+
+   ```bash
+   cd $SDE_INSTALL/bin/
+   ./vfio_bind.sh 8086:1453
+   Device: 0000:00:01.6 Group = 5
+   ```
+
+   Value of `iommu_grp_num` should be `5`
+
+- `vport`
+
+   Number of vports supported are from [0-6].
+   For example: `vport=[0-1]`
+
+- `eal_args`
+
+   Support values for `--proc-type` are `primary` and `auto`
+   Note: In case of multi-process setup which is supported in docker
+   environment, --proc-type can be used to specify the process type.
+
+- `cfgqs-idx`
+
+   We give options to each process (primary or secondary) to request
+   numbers of configure Queues. Admin must set cfgqs-idx between `"0-15"`,
+   recommended option when running only 1 process. Plan and split config
+   queues between multi-process. For example, to configure one cfgq; use
+   `cfgqs-idx: "1"`. Supported index numbers are from 0 to 15.
+
+- Specify the correct `program-name` and `p4_pipeline_name`
+
+- Specify the absolute paths for `bfrt-config`, `context` and `config`.
+
+### 3.2 Prepare the runtime binary
+
+On ACC, use TDI pipeline builder to combine the artifacts generated by the
+`p4c-pna-xxp` compiler and generate a runtime binary
+
+```bash
+$P4CP_INSTALL/bin/tdi_pipeline_builder \
+    --p4c_conf_file=/usr/share/stratum/es2k_skip_p4.conf \
+    --bf_pipeline_config_binary_file=$OUTPUT_DIR/simple_l3_l4_pna.pb.bin
```

Comment:
Here we are referring to "simple_l3_l4_pna" program, but above in this document we refer "simple_l2_demo"